### PR TITLE
Add an iOS target

### PR DESCRIPTION
**Pull request by gfontenot**
This adds an iOS framework target to the project, along with an iOS test
bundle. This supersedes #14.

### CODE REVIEW

File: Box.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -162,11 +204,47 @@
...
+				F8BB81CF1A939B66001AA352 /* Frameworks */,
+				F8BB81D01A939B66001AA352 /* Headers */,
+				F8BB81D11A939B66001AA352 /* Resources */,
+			);
+			buildRules = (
+			);
+			dependencies = (
+			);
+			name = "Box-iOS";
+			productName = "Box-iOS";
```
Comment (robrix) [inline Box.xcodeproj/project.pbxproj]: Can the product for both the Mac and iOS frameworks just be `Box`?

File: Box.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -401,6 +526,90 @@
...
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				COPY_PHASE_STRIP = NO;
+				FRAMEWORK_SEARCH_PATHS = (
+					"$(SDKROOT)/Developer/Library/Frameworks",
+					"$(inherited)",
+				);
+				INFOPLIST_FILE = BoxTests/Info.plist;
+				IPHONEOS_DEPLOYMENT_TARGET = 8.0;
+				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
+				PRODUCT_NAME = "$(TARGET_NAME)";
```
Comment (robrix) [inline Box.xcodeproj/project.pbxproj]: I guess this is the setting we’d want to change for that.

File: Box.xcodeproj/xcshareddata/xcschemes/Box-Mac.xcscheme [hunk 3]
```diff
@@ -16,21 +16,21 @@
...
             </BuildableReference>
          </BuildActionEntry>
          <BuildActionEntry
             buildForTesting = "YES"
-            buildForRunning = "NO"
+            buildForRunning = "YES"
             buildForProfiling = "NO"
             buildForArchiving = "NO"
-            buildForAnalyzing = "NO">
+            buildForAnalyzing = "YES">
```
Comment (robrix) [inline Box.xcodeproj/xcshareddata/xcschemes/Box-Mac.xcscheme]: Please disable building the tests for analyzing and for running.

File: Box.xcodeproj/xcshareddata/xcschemes/Box-iOS.xcscheme [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+               BlueprintName = "Box-iOS"
+               ReferencedContainer = "container:Box.xcodeproj">
+            </BuildableReference>
+         </BuildActionEntry>
+         <BuildActionEntry
+            buildForTesting = "YES"
+            buildForRunning = "YES"
+            buildForProfiling = "NO"
+            buildForArchiving = "NO"
+            buildForAnalyzing = "YES">
```
Comment (robrix) [inline Box.xcodeproj/xcshareddata/xcschemes/Box-iOS.xcscheme]: Ditto for the iOS scheme.

**Comment by robrix** [comment]
Just a couple of notes. This looks great, thanks!

File: Box.xcodeproj/project.pbxproj (hunk 1 above)
Comment (gfontenot) [inline Box.xcodeproj/project.pbxproj]: What the hell. I could have sworn I did this.

File: Box.xcodeproj/project.pbxproj (hunk 2 above)
Comment (gfontenot) [inline Box.xcodeproj/project.pbxproj]: I think this is for (one of) the test bundles. I can't find this string anywhere else in the build settings.

**Comment by gfontenot** [comment]
@robrix Should be all set now.

File: Box.xcodeproj/project.pbxproj (hunk 2 above)
Comment (robrix) [inline Box.xcodeproj/project.pbxproj]: Ah, so it is—my bad!

**Comment by robrix** [comment]
Thanks so much! :bow:

**Pull request merged**